src/api/session.ts

```diff
@@ -22,6 +23,7 @@ export class SessionAPI {
     public readonly outgoingFriendRequests: ComputedRef<User[]>;
     public readonly incomingFriendRequests: ComputedRef<User[]>;
     public readonly friends: ComputedRef<User[]>;
+    public readonly libraryPage: Ref<MapData | null> = shallowRef(null);
```

Comment:
I was unsure the best way to implement this - this is meant to be for navigation data for loading up specific library pages. I named it this way with the intent of it being usable for the eventual unit pages, but there is probably a better implementation.

Reply:
The best way to do this is by passing query/data to the route, i.e. https://router.vuejs.org/guide/essentials/passing-props.html#object-mode

_However_, this currently isn't compatible with the automatically generated routes by the `vue-auto-routing` package that this project is currently using, so actually the way you've done it here is actually the only way I can think to do it, but it is indeed quite hacky. I'm currently in the process of switching the project over to Vite and this newer auto routing package: https://github.com/posva/unplugin-vue-router, which will allow us to use route params properly. I'll remember to fix this once that's done